lib/pure/asyncfutures.nim

```diff
@@ -30,6 +30,47 @@ type
 when not defined(release):
   var currentID = 0
 
+const isFutureLoggingEnabled* = defined(futureLogging)
+when isFutureLoggingEnabled:
+  import hashes
+  type
+    FutureInfo = object
+      stackTrace: seq[StackTraceEntry]
+      fromProc: string
+
+  var futuresInProgress {.threadvar.}: Table[FutureInfo, int]
+
+  proc getFuturesInProgress*(): var Table[FutureInfo, int] =
```

Comment:
This whole proc is not necessary with the new table implementation.

Reply:
To be honest it's probably better to have this proc anyway. Accessing a global var doesn't seem like great API design.

Reply:
But you don't export it so it's not "API design".

Reply:
What? The proc is exported

Reply:
You should use a TableRef then with this logic, this ``var Table`` here is fragile :P

Reply:
Huh, can you explain why it is fragile?